search/elastic.py

```diff
@@ -341,47 +342,73 @@ def field_property(field_name, field_value):
+                self._check_mappings(doc_type, source)
+                id_ = source['id'] if 'id' in source else None
+                log.debug("indexing %s object with id %s", doc_type, id_)
+                action = {
+                    "_index": self.index_name,
+                    "_type": doc_type,
+                    "_id": id_,
+                    "_source": source
+                }
+                actions.append(action)
+            # bulk() returns a tuple with summary information
+            # number of successfully executed actions and number of errors if stats_only is set to True.
+            _, num_errors = bulk(
+                self._es,
+                actions,
+                **kwargs
             )
-        except exceptions.ElasticsearchException as ex:
+            if num_errors:
+                for num_error in num_errors:
+                    log.exception("error while indexing - %s", num_error.message)
+                    raise num_error
+        # Broad exception handler to protect around bulk call
+        except Exception as ex:
             # log information and re-raise
-            log.exception("error while indexing - %s", ex.message)
+            log.exception("error while deleting document from index - %s", ex.message)
             raise ex
 
-    def remove(self, doc_type, doc_id, **kwargs):
-        """ Implements call to remove the document from the index """
-
-        log.debug("remove index for %s object with id %s", doc_type, doc_id)
+    def remove(self, doc_type, doc_ids, **kwargs):
+        """ Implements call to remove the documents from the index """
 
         try:
             # ignore is flagged as an unexpected-keyword-arg; ES python client documents that it can be used
             # pylint: disable=unexpected-keyword-arg
-            self._es.delete(
-                index=self.index_name,
-                doc_type=doc_type,
-                id=doc_id,
+            actions = []
+            for doc_id in doc_ids:
+                log.debug("remove index for %s object with id %s", doc_type, doc_id)
+                action = {
+                    '_op_type': 'delete',
+                    "_index": self.index_name,
+                    "_type": doc_type,
+                    "_id": doc_id
+                }
+                actions.append(action)
+            # bulk() returns a tuple with summary information
+            # number of successfully executed actions and number of errors if stats_only is set to True.
+            _, num_errors = bulk(
+                self._es,
+                actions,
                 # let notfound not cause error
                 ignore=[404],
                 **kwargs
             )
-        except exceptions.ElasticsearchException as ex:
+            if num_errors:
+                for num_error in num_errors:
+                    log.exception("error while indexing - %s", num_error.message)
+                    raise num_error
```

Comment:
Sorry, I'm confused by what's going on here.

1. What is the type of `num_errors`? I thought it was an integer before, but now it looks like it's an enumerable of some sort?
2. If there are multiple errors, won't this stop at the first one and hide all the others (because you're doing a `raise` in the loop)?